libp2p.nimble

```diff
@@ -16,19 +16,43 @@ requires "nim >= 1.2.0",
          "secp256k1",
          "stew"
 
-proc runTest(filename: string, secure: string = "secio") =
-  exec "nim c -r --opt:speed -d:debug --verbosity:0 --hints:off tests/" & filename
+proc runTest(filename: string, secure: string = "secio", verify: bool = true, sign: bool = true) =
+  var excstr: string = "nim c -r --opt:speed -d:debug --verbosity:0 --hints:off"
```

Comment:
meh - recompiling just to test different runtime flags? if the gossipsub cleanup is so broken that we need a fresh process for each run, these flags should at least be runtime flags to the testgossip executable, because they are runtime flags in the library

Reply:
btw, are there tests that verify that an unsigned message is discarded by a signed switch?

Reply:
> meh - recompiling just to test different runtime flags? if the gossipsub cleanup is so broken that we need a fresh process for each run, these flags should at least be runtime flags to the testgossip executable, because they are runtime flags in the library

Yeah, runtime flags would be ideal, but we need to rewrite our tests for that, shouldn't be too much work, but this seemed quickier so opted for it for now. Note that the whole `StandardSwitch` construct should be superseded by #133 sometimes soon. In other words, this is just to get us by right now.

Reply:
> btw, are there tests that verify that an unsigned message is discarded by a signed switch?

Not right now and we should add them - @sinkingsugar could you take a look at that please? The plus side is that we know that it works at least right now, since the point of the PR is to prevent unsigned messages from being dropped, but we should add the full battery of tests for sure.